linux/src/threadpool_linux.c

```diff
@@ -811,25 +797,20 @@ int threadpool_schedule_work_item(THANDLE(THREADPOOL) threadpool, THREADPOOL_WOR
             }
             else
             {
+                THREADPOOL_WORK_ITEM_CONTEXT_HANDLE threadpool_work_item_context_ptr = THANDLE_GET_T(THREADPOOL_WORK_ITEM_CONTEXT)(threadpool_work_item_context);
                 /* Codes_SRS_THREADPOOL_LINUX_05_020: [ threadpool_schedule_work_item shall acquire the SRW lock in shared mode by calling srw_lock_acquire_exclusive. ]*/
                 srw_lock_acquire_shared(threadpool_ptr->srw_lock);
-                THREADPOOL_TASK* task_item = &threadpool_ptr->task_array[insert_pos];
-
-                (void)interlocked_increment(&threadpool_work_item->pending_work_item_count);                
+                THREADPOOL_TASK* task_item = &threadpool_ptr->task_array[insert_pos];     
 
-                /* Codes_SRS_THREADPOOL_LINUX_05_022: [ threadpool_schedule_work_item shall copy the work_function and work_function_context from threadpool_work_item into insert position in the task array. ]*/
-                task_item->work_function_ctx = threadpool_work_item->work_function_ctx;
-                task_item->work_function = threadpool_work_item->work_function;
+                /* Codes_SRS_THREADPOOL_LINUX_05_022: [ threadpool_schedule_work_item shall copy the work_function and work_function_context from threadpool_work_item_context into insert position in the task array. ]*/
+                task_item->work_function_ctx = threadpool_work_item_context_ptr->work_function_ctx;
+                task_item->work_function = threadpool_work_item_context_ptr->work_function;
 
                 /* Codes_SRS_THREADPOOL_LINUX_05_023: [ threadpool_schedule_work_item shall set the task_state to TASK_WAITING and then release the shared SRW lock by calling srw_lock_release_exclusive. ]*/
                 (void)interlocked_exchange(&task_item->task_state, TASK_WAITING);               
```

Comment:
>                [](http://example.com/codeflow?start=81&length=15)

nit: consider installing this extension in VS to get rid of trailing spaces [https://marketplace.visualstudio.com/items?itemName=MadsKristensen.TrailingWhitespace64](https://marketplace.visualstudio.com/items?itemName=MadsKristensen.TrailingWhitespace64)

Reply:
Okay, this one , I am using VS code for Linux, so might have not been installed.